src/lib/Event/Listener/LoginListener.php

```diff
@@ -75,16 +72,31 @@ public function onSecurityInteractiveLogin(InteractiveLoginEvent $event): void
             return;
         }
 
+        $currentSiteAccess = $this->siteAccessService->getCurrent();
+
+        if ($currentSiteAccess === null) {
+            return;
+        }
+
+        $siteAccessName = $currentSiteAccess->name;
+        $endpoint = $this->getEndpoint($siteAccessName);
+        $customerId = $this->getCustomerId($siteAccessName);
+
+        if (!isset($customerId, $endpoint)) {
+            return;
+        }
+
         if (!$event->getRequest()->cookies->has(RecommendationSession::RECOMMENDATION_SESSION_KEY)) {
             if (!$this->session->isStarted()) {
                 $this->session->start();
             }
             $event->getRequest()->cookies->set(RecommendationSession::RECOMMENDATION_SESSION_KEY, $this->session->getId());
         }
 
-        $notificationUri = sprintf($this->getNotificationEndpoint() . '%s/%s/%s',
-            'login',
-            $event->getRequest()->cookies->get(RecommendationSession::RECOMMENDATION_SESSION_KEY),
+        $notificationUri = $this->getNotificationUri(
+            $endpoint,
+            $customerId,
+            (string) $event->getRequest()->cookies->get(RecommendationSession::RECOMMENDATION_SESSION_KEY),
```

Comment:
`RecommendationSession::RECOMMENDATION_SESSION_KEY` might be stored as a shorter variable, given it's been used several times.